### PR TITLE
add changelog

### LINKED ISSUE #452: Introduce change log
I propose to keep a changelog to make things easier for fellow developers/users when we release a new version. All notable changes to this project should be documented in a change log file.

I usually base mine off the "[Keep a Changelog](https://keepachangelog.com/en/1.0.0)" format. There's not really a reason to follow semver for this project, so I'd recommend just removing that part of the template.

/cc @ysmolsky @tomponline

### PR DESCRIPTION
**Pull request by ysmolsky**
Fixes #452

### CODE REVIEW

**Comment by ysmolsky** [comment]
This is the beginning of the changelog to simplify releases. Whenever we do something important in the project we should add one line to this file. This can be used for release notes later.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+All notable changes to this project will be documented in this file.
+
+## [Unreleased]
+### Added
+- Support of UNIX domain sockets
+- Solaris/illumos support
+- the "reserve-job" command 
+- draining status to the "stats" command
+
+### Changed
```
Comment (tomponline) [inline CHANGELOG.md:11]: @ysmolsky we should probably add the default sync interval change here too.

Comment (ysmolsky) [reply CHANGELOG.md:12]: added!

**Comment by ysmolsky** [comment]
I am still not sure if we need to have two section: Added/Changed. Or should we rather just copy important commits as they are without having these two sections. Right now we have this mixed, in 1.11 and 1.10 I used just copy-commit-msg style and for the unreleased I edited them by hand. I don't want to produce unnecessary friction here.

**Comment by tomponline** [comment]
@ysmolsky I dont mind, what ever is easier :)

**Pull request merged**